src/memray/reporters/stats.py

```diff
@@ -202,21 +131,25 @@ def render(
         for entry in self._get_top_allocations_by_count():
             print(f"\t- {entry}")
 
-    def _get_stats_data(self) -> _StatsData:
-        return get_stats_data(self.data)
-
-    def _get_top_allocations_by_size(self) -> Generator[str, None, None]:
-        yield from get_top_allocations_by_size(self.data, self.num_largest)
-
-    def _get_top_allocations_by_count(self) -> Generator[str, None, None]:
-        yield from get_top_allocations_by_count(self.data, self.num_largest)
-
-    def _get_allocator_type_distribution(
-        self, alloc_type_counter: Dict[str, int]
-    ) -> Generator[str, None, None]:
-        yield from get_allocator_type_distribution(alloc_type_counter)
-
-    def _draw_histogram(
-        self, data: List[int], bins: int, *, hist_scale_factor: int = 25
-    ) -> str:
-        return draw_histogram(data, bins, hist_scale_factor=hist_scale_factor)
+    @staticmethod
+    def _format_location(loc: Tuple[str, str, int]) -> str:
+        function, file, line = loc
+        if function == "<unknown>":
+            return "<stack trace unavailable>"
+        return f"{function}:{file}:{line}"
+
+    def _get_top_allocations_by_size(self) -> Iterator[str]:
+        for location, size in self._stats.top_locations_by_size:
+            yield f"{self._format_location(location)} -> {size_fmt(size)}"
+
+    def _get_top_allocations_by_count(self) -> Iterator[str]:
+        for location, count in self._stats.top_locations_by_count:
+            yield f"{self._format_location(location)} -> {count}"
+
+    def _get_allocator_type_distribution(self) -> Iterator[str]:
+        for allocator_name, count in sorted(
+            self._stats.allocation_count_by_allocator.items(),
+            key=lambda item: item[1],
```

Comment:
Nit: you can use `operator.itemgetter` for better performance

Reply:
This dict contains one entry per allocator, and we've got less than 16 of them. I'm gonna keep the more readable, less performant one, simply because this dict is too small to care.